src/components/DropZone/components/FileUpload/FileUpload.tsx

```diff
@@ -101,35 +63,49 @@ export function FileUpload(props: FileUploadProps) {
     </div>
   );
 
-  const mediumView =
-    size === 'medium' ? (
-      <Stack vertical spacing="tight">
-        {actionTitleMarkup}
-        <Caption>
-          <TextStyle variation="subdued">{actionHint}</TextStyle>
-        </Caption>
-      </Stack>
-    ) : null;
-
-  const smallView =
-    size === 'small' ? (
-      <Stack vertical spacing="tight">
-        <Icon source={DragDropMajorMonotone} color="inkLightest" />
-      </Stack>
-    ) : null;
-
   const fileUploadClassName = classNames(
     styles.FileUpload,
     newDesignLanguage && styles.newDesignLanguage,
     measuring && styles.measuring,
+    size === 'small' && styles.FileUploadSmallView,
   );
 
-  return (
-    <div className={fileUploadClassName}>
-      {smallView}
-      {mediumView}
-      {largeView}
-      {extraLargeView}
-    </div>
-  );
+  let viewMarkup;
+  switch (size) {
+    case 'extraLarge':
+      viewMarkup = (
+        <Stack vertical>
+          <img width="40" src={uploadArrow} alt="" />
+          {buttonMarkup}
+          <TextStyle variation="subdued">{actionHint}</TextStyle>
+        </Stack>
+      );
+      break;
+    case 'large':
+      viewMarkup = (
+        <Stack vertical spacing="tight">
+          <img width="40" src={uploadArrow} alt="" />
+          {buttonMarkup}
+          <Caption>
+            <TextStyle variation="subdued">{actionHint}</TextStyle>
+          </Caption>
+        </Stack>
+      );
+      break;
+    case 'medium':
+      viewMarkup = (
+        <Stack vertical spacing="tight">
+          {actionTitleMarkup}
+          <Caption>
+            <TextStyle variation="subdued">{actionHint}</TextStyle>
+          </Caption>
+        </Stack>
+      );
+      break;
+    case 'small':
+      viewMarkup = <img width="20" src={uploadArrow} alt="" />;
+      break;
+  }
```

Comment:
This markup before had four different views being defined and rendered as null if it isn't the correct size. This moves the logic into one spot.